app/controllers/CheckYourAnswersController.scala

```diff
@@ -157,31 +159,60 @@ class CheckYourAnswersController @Inject()(
   def onSubmit(): Action[AnyContent] = (identify andThen notEnrolled andThen getData andThen requireData).async {
     implicit request =>
 
-      taxEnrolmentsConnector.createSubscription(request.userAnswers).flatMap {
-        subscriptionResponse =>
-          if (subscriptionResponse.status.equals(NO_CONTENT)) {
-           emailService.sendEmail(request.userAnswers).map {
-             emailResponse =>
-               logEmailResponse(emailResponse)
-                Redirect(routes.RegistrationSuccessfulController.onPageLoad())
-           }.recover {
-             case e: Exception => Redirect(routes.RegistrationSuccessfulController.onPageLoad())
-           }
-          } else {
-            Future(InternalServerError("ERROR PAGE TO GO HERE"))
-
+          (request.userAnswers.get(DoYouHaveUTRPage), request.userAnswers.get(RegistrationTypePage),
+            request.userAnswers.get(DoYouHaveANationalInsuranceNumberPage)) match {
+
+            case (Some(true), None, None) => createEnrolment(request.userAnswers) // TODO: EIS subscription for business with ID
+            case (Some(false), Some(Individual), Some(true)) => createEnrolment(request.userAnswers) // TODO: EIS subscription for individual with ID
+
+            case (Some(false), Some(Individual), Some(false)) => registrationService.sendIndividualRegistration(request.userAnswers) flatMap {
+              case Some(response) => response.status match {
+                case OK => createEnrolment(request.userAnswers)
+                case _ => Future.successful(Redirect(UnsuccessfulSubscriptionController.onPageLoad()))
+              }
+              case _ => Future.successful(Redirect(UnsuccessfulSubscriptionController.onPageLoad()))
+            }
+
+            case (Some(false), Some(Business), None) =>
+              registrationService.sendOrganisationRegistration(request.userAnswers) flatMap {
+                case Some(response) => response.status match {
+                  case OK => createEnrolment(request.userAnswers)
+                  case _ => Future.successful(Redirect(UnsuccessfulSubscriptionController.onPageLoad()))
+                }
+                case _ => Future.successful(Redirect(UnsuccessfulSubscriptionController.onPageLoad()))
+              }
+
+            case _ => Future.successful(Redirect(UnsuccessfulSubscriptionController.onPageLoad()))
           }
+
       }
-  }
 
+      def logEmailResponse(emailResponse: Option[HttpResponse]): Unit = {
```

Comment:
put the private back on, it's not going to be an endpoint. Utility methods should be private